src/ClientImpl.cpp

```diff
@@ -644,6 +784,72 @@ class SyncCallback : public ProcedureCallback {
     InvocationResponse *m_responseOut;
 };
 
+
+/* compares value first timestamp with second timestamp and returns
+ * 0 if equal,
+ * 1 if first is greater or
+ * -1 if first is less thn second
+
+ */
+int compareTimeVal(const struct timeval& first, const struct timeval& second) {
+    if (first.tv_sec < second.tv_sec) {
+        return -1;
+    }
+    if (first.tv_sec > second.tv_sec) {
+        return 1;
+    }
+    if (first.tv_usec < second.tv_usec) {
+        return -1;
+    }
+    if (first.tv_usec > second.tv_usec) {
+        return 1;
+    }
+    return 0;
+}
+
+bool isTimeValLessThn (const struct timeval& first, const struct timeval& second) {
+    return compareTimeVal(first, second) < 0;
+}
+
+void ClientImpl::purgeExpiredRequests() {
+    struct timeval now;
+    event_base_gettimeofday_cached(m_base, &now);
+    BEVToCallbackMap::iterator end = m_callbacks.end();
+
+    std::vector<voltdb::Table> dummyTable;
+    InvocationResponse response(0, voltdb::STATUS_CODE_CONNECTION_TIMEOUT, "client timedout waiting for response",
+            voltdb::STATUS_CODE_UNINITIALIZED_APP_STATUS_CODE, "No response received in allotted time",
+            dummyTable);
+
+    for (BEVToCallbackMap::iterator itr = m_callbacks.begin(); itr != end; itr++) {
+        boost::shared_ptr<CallbackMap> callbackMap = itr->second;
+        for (CallbackMap::iterator cbItr =  callbackMap->begin();
+                cbItr != callbackMap->end(); ++cbItr) {
+            timeval expirationTime = cbItr->second->getExpirationTime();
+
+            if (cbItr->second->isReadOnly() && (compareTimeVal(expirationTime, now) <= 0)) {
+                response.setClientData(cbItr->first);
+                try {
+                    cbItr->second->getCallback()->callback(response);
+                } catch (std::exception &excp) {
+                    if (m_listener.get() != NULL) {
+                        try {
+                            m_listener->uncaughtException(excp, cbItr->second->getCallback(), response);
+                        } catch (std::exception &e) {
+                            std::ostringstream msg;
```

Comment:
maybe would be cheaper and simpler to have it this way?:
std::string msg("Uncaught exception. ");
logMessage(ClientLogger::ERROR, msg+e.what());